build.gradle

```diff
@@ -715,6 +735,11 @@ project(':iceberg-parquet') {
 }
 
 project(':iceberg-arrow') {
+    test {
+        if (JavaVersion.current() == JavaVersion.VERSION_17) {
+            jvmArgs += ["-Xmx1024m", "-XX:+IgnoreUnrecognizedVMOptions", "--add-opens", "java.base/java.lang.invoke=ALL-UNNAMED", "--add-opens", "java.base/java.io=ALL-UNNAMED", "--add-opens", "java.base/java.lang.reflect=ALL-UNNAMED", "--add-opens", "java.base/java.lang=ALL-UNNAMED", "--add-opens", "java.base/java.math=ALL-UNNAMED", "--add-opens", "java.base/java.net=ALL-UNNAMED", "--add-opens", "java.base/java.nio=ALL-UNNAMED", "--add-opens", "java.base/java.text=ALL-UNNAMED", "--add-opens", "java.base/java.time=ALL-UNNAMED", "--add-opens", "java.base/java.util.concurrent.atomic=ALL-UNNAMED", "--add-opens", "java.base/java.util.concurrent=ALL-UNNAMED", "--add-opens", "java.base/java.util.regex=ALL-UNNAMED", "--add-opens", "java.base/java.util=ALL-UNNAMED", "--add-opens", "java.base/jdk.internal.ref=ALL-UNNAMED", "--add-opens", "java.base/jdk.internal.reflect=ALL-UNNAMED", "--add-opens", "java.sql/java.sql=ALL-UNNAMED"]
```

Comment:
Can you explain why these are needed?

Reply:
Arrow test were failing without these flags, for ex: 

```
org.apache.iceberg.arrow.vectorized.ArrowReaderTest > testHasNextIsIdempotent FAILED
    java.lang.NoClassDefFoundError: Could not initialize class org.apache.arrow.memory.util.MemoryUtil
 at org.apache.arrow.memory.util.MemoryUtil.<clinit>(MemoryUtil.java:138)
            ... 16 more

            Caused by:
            java.lang.reflect.InaccessibleObjectException: Unable to make field long java.nio.Buffer.address accessible: module java.base does not "opens java.nio" to unnamed module @99e937b
```

hence it required jdk flags like `--add-opens=java.base/java.io=ALL-UNNAMED` hence added the same here.

Reply:
Can you add newlines?

Reply:
ACK

Reply:
Done